### ISSUE TITLE
project resource should not store environment information

### ISSUE THREAD

**Issue by bschaeffer** [launchdarkly/terraform-provider-launchdarkly#13]
If I can't manage the environments directly inside a `launchdarkly_project` resource, why is it nesting them inside the resource in the state file? 

You already provide `launchdarkly_environment` resource, which is the only means I have of actually modifying and tracking environment attributes, so I think essentially tracking environment data twice is superfluous.

For instance, take the following:

```hcl
resource "launchdarkly_project" "default" {
  key  = "default"
  name = "default"
}

resource "launchdarkly_environment" "staging" {
  project_key = launchdarkly_project.default.key
  name        = "staging"
  key         = "staging"
}
```

This results in tracking the environment data twice in the statefile:

```bash
$ terrraform show
# launchdarkly_environment.staging
resource "launchdarkly_environment" "staging" {
    id                   = "default/staging"
    key                  = "staging"
    mobile_key           = (sensitive value)
    name                 = "staging"
    project_key          = "default"
}

# launchdarkly_project.default:
resource "launchdarkly_project" "default" {
    id   = "default"
    key  = "default"
    name = "default"

    environments {
        key  = "staging"
        name = "staging"
    }
}
```

I think it would be better if the project resource just tracked project attributes only and let environment resources be explicitly defined in the terraform.

**Comment by ldhenry** [comment]
To give some background on this:
1. When creating a new project using the LaunchDarkly API, `test` and `production` environments are created in the same call. At the time, it seemed a little clunky to immediately delete these environments to remove the need for nested environments in the `launchdarkly_project_resource`.
2. The AWS provider's way of handling [inline and discrete security groups](http://cavaliercoder.com/blog/inline-vs-discrete-security-groups-in-terraform.html) served as prior art to influence the support for the ability declare environments nested under the `launchdarkly_project` resource or as discrete `launchdarkly_environment` resources. In fact it is currently possible to manage environments in the `launchdarkly_project` resource but it is not documented.

To make a long story short, after spending some time using the provider I came to a similar conclusion as you that the `launchdarkly_project` resource should not store environment information. I plan to deprecate this functionality in the next release.

Thanks again for your feedback,
Henry

**Comment by ldhenry** [comment]
Hey @bschaeffer,

Better late than never on this one I suppose. We figured out an acceptable way to handle nested environments in the `launchdarkly_project` resource and implemented it in [v1.5.0](https://github.com/launchdarkly/terraform-provider-launchdarkly/releases/tag/v1.5.0). One of the things we had to do to get it working was to remove the `computed` property from the `environments` block. This means that once you upgrade to v1.5.0, and run a `terraform apply`, the environment information stored in the `launchdarkly_project` resource will be removed from the state(if you are not using nested environments).

Hope it helps,
Henry
